mesonbuild/interpreter.py

```diff
@@ -3114,7 +3134,7 @@ def func_find_program(self, node, args, kwargs):
         for_machine = self.machine_from_native_kwarg(kwargs)
         return self.find_program_impl(args, for_machine, required=required,
                                       silent=False, wanted=wanted,
-                                      search_dirs=search_dirs)
+                                      search_dirs=search_dirs, fallback=kwargs.get('fallback', None))
```

Comment:
I would pass the whole kwargs, that you can pass down to dependency_fallback() instead of creating a new kwarg dict there, because you'll need to also pass down `default_options` for example.